### PR TITLE
fix: Read and store UI variables after the UI has loaded

### PR DESCRIPTION
**Pull request by aliasliao**
PR https://github.com/mdn/to-do-notifications/pull/22 causes a runtime error and makes the app completely unusable.
This pr just reverts it.

image

### CODE REVIEW

**Review by teoli2003 — CHANGES_REQUESTED** [review]
This is not the correct fix. The correct fix is to read the UI elements, and performing the `Notification` test, inside the `window.onload` handler rather than before.

Do you want to do the change?

File: scripts/todo.js [hunk 1]
```diff
@@ -1,42 +1,35 @@
...
 
-// create a reference to the notifications list in the bottom of the app; we will write database messages into this list by
-//appending list items as children of this element
-const note = document.getElementById('notifications');
-
 // create an instance of a db object for us to store the IDB data in
 let db;
 
-// create a blank instance of the object that is used to transfer data into the IDB. This is mainly for reference
-const newItem = [
```
Comment (aliasliao) [inline scripts/todo.js:11]: unused reference

File: scripts/todo.js [hunk 2]
```diff
@@ -1,42 +1,35 @@
...
 
-const hours = document.getElementById('deadline-hours');
-const minutes = document.getElementById('deadline-minutes');
-const day = document.getElementById('deadline-day');
-const month = document.getElementById('deadline-month');
-const year = document.getElementById('deadline-year');
+  const taskForm = document.getElementById('task-form');
+  const title = document.getElementById('title');
 
-const submit = document.getElementById('submit');
```
Comment (aliasliao) [inline scripts/todo.js:27]: unused reference

**Comment by aliasliao** [comment]
> This is not the correct fix. The correct fix is to read the UI elements, and performing the `Notification` test, inside the `window.onload` handler rather than before.
> 
> Do you want to do the change?

okay. Have committed the correct changes.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (teoli2003) [inline .gitignore:1]: This change shouldn't be in this PR.

File: index.html [hunk 4]
```diff
@@ -107,5 +106,6 @@ <h2>Add new to do.</h2>
   </button>
 </div>
 
+  <script src="scripts/todo.js"></script>
```
Comment (teoli2003) [inline index.html:109]: We can revert this change now

**Review by teoli2003 — CHANGES_REQUESTED** [review]
I've added a commit to clean the comments (that were sometimes outdated).
I've also fixed a bug in the `ordinal` function and moved more things inside the load handler.

There are a few things to revert (.gitignore, and the position of `<script>` in the HTML.

After that, I think we should merge this and do follow-ups to fix problems we would have missed.

Great to see this being fixed.

**Review by teoli2003 — APPROVED** [review]
Let's merge this. Follow-ups in other PRs.

**Comment by teoli2003** [comment]
Thanks a lot!

Looks good to me.

Congratulations upon your first merged commit here! Welcome aboard! :tada:

**Pull request merged**